### PR TITLE
Adding the ability to create DAI locks

### PR DESCRIPTION
**Pull request by julien51**
# Description

This is for now the onluy ERC20 token that we support on the UI for lock creation.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [X] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [X] My code follows the style guidelines of this project, including naming conventions
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] If my code adds or changes components, I have written corresponding stories with Storybook
- [X] I have performed a self-review of my own code
- [X] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Screenshots:
image

image


When "editing" an existing lock (we cannot change the currency!)
image

File: unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js [hunk 1]
```diff
@@ -159,6 +163,54 @@ describe('CreatorLockForm', () => {
     })
   })
 
+  describe('currency', () => {
+    it('should default to Ether', () => {
+      expect.assertions(1)
+      const wrapper = makeLockForm()
+      expect(
+        wrapper.getByText(`Use ${config.ERC20Contract.name}`)
```
Comment (cellog) [inline unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js]: maybe add a comment that this is asserting on the link to change the currency. I stared at this for a solid minute trying to figure out why this was a valid test :)_

File: unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js [hunk 2]
```diff
@@ -159,6 +163,54 @@ describe('CreatorLockForm', () => {
...
+      ).not.toBeNull()
+    })
+
+    it('should handle toggling of the currency', () => {
+      expect.assertions(2)
+      const wrapper = makeLockForm()
+
+      const swicthToErc20Link = wrapper.getByText(
```
Comment (cellog) [inline unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js]: typo `swicth...`

File: unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js [hunk 3]
```diff
@@ -159,6 +163,54 @@ describe('CreatorLockForm', () => {
...
+      const switchToEther = wrapper.getByText('Use Ether')
+      expect(switchToEther).not.toBeNull()
+
+      rtl.fireEvent.click(switchToEther)
+      expect(
+        wrapper.getByText(`Use ${config.ERC20Contract.name}`)
+      ).not.toBeNull()
+    })
+
+    it('should create a lock with the picked currency', () => {
```
Comment (cellog) [inline unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js]:
```suggestion
    it('should create a lock with the chosen currency', () => {
```

File: unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js [hunk 4]
```diff
@@ -159,6 +163,54 @@ describe('CreatorLockForm', () => {
...
+      expect.assertions(2)
+      const wrapper = makeLockForm()
+
+      const swicthToErc20Link = wrapper.getByText(
```
Comment (cellog) [inline unlock-app/src/__tests__/components/creator/CreatorLockForm.test.js]: same typo here

**Pull request merged**